Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c

```diff
@@ -117,6 +118,11 @@ FillBootOptionListFromCfgData (
       // LBA should be defined as 64bit.
       // Will remove the typecast when the structure is fixed.
       BootOption->Image[ImageIdx].LbaImage.LbaAddr = (UINT32)Lba;
+    } else if ((StrPtr[0] == '0') && ((StrPtr[1] == 'x') || (StrPtr[1] == 'X')) &&
+               (AsciiStrHexToUint64S (StrPtr, NULL, &MemoryAddress) == RETURN_SUCCESS)) {
+      // For Boot from Memory case the container image address must be like 0x01234567.
+      // Here the LbaAddress is reused as container image address.
+      BootOption->Image[ImageIdx].LbaImage.LbaAddr = (UINT32)MemoryAddress;
```

Comment:
A hex value could be used as a file name.  Line 114 asked LBA address to start with "#" to reduce conflict risk. Since we already defined it, could we re-use it instead of adding this new condition?